biz.aQute.bndlib/src/aQute/bnd/osgi/resource/CapReq.java

```diff
@@ -20,26 +22,27 @@
 	private final Resource				resource;
 	private final Map<String,String>	directives;
 	private final Map<String,Object>	attributes;
+	private transient int				hashCode	= 0;
```

Comment:
This is the only mutable field in the class. If we build it in the constructor instead of lazily, this class can be marked `@Immutable`